### PR TITLE
Remove unused locals in System.Security namespaces

### PR DESCRIPTION
**Pull request by MattKotsenas**
Remove unused locals in the System.Security namespaces.

Fixes a part of #30457.

### CODE REVIEW

File: src/libraries/System.Security.AccessControl/src/System/Security/AccessControl/ACL.cs [hunk 1]
```diff
@@ -587,7 +587,6 @@ public void InsertAce(int index, GenericAce ace)
 
         public void RemoveAce(int index)
         {
-            GenericAce ace = _aces[index];
             _aces.RemoveAt(index);
```
[the remark this answers is not in the document]
Comment (MattKotsenas) [reply src/libraries/System.Security.AccessControl/src/System/Security/AccessControl/ACL.cs:590]: Thanks @bartonjs! There's a test for -1, but not a value too large, so I added a new case to the existing unit test to also try calling with `int.MaxValue`.

Let me know if that isn't what you intended, or if you have any questions. Thanks!

[the remark this answers is not in the document]
Comment (bartonjs) [reply src/libraries/System.Security.AccessControl/src/System/Security/AccessControl/ACL.cs:590]: Just to be clear: You ran the test both with and without removing this line and the exception was the same?

File: src/libraries/System.Security.AccessControl/tests/RawAcl/RawAcl_RemoveAce.cs [hunk 2]
```diff
@@ -41,6 +41,13 @@ public static void TestRemoveAce(string sddl)
                 rawAcl.RemoveAce(index);
             });
 
+            //test remove at value too large
+            Assert.Throws<ArgumentOutOfRangeException>(() =>
```
Comment (bartonjs) [inline src/libraries/System.Security.AccessControl/tests/RawAcl/RawAcl_RemoveAce.cs:45]: Assuming there's a paramName being specified:

```suggestion
            AssertExtensions.Throws<ArgumentOutOfRangeException>(
                "whateverTheParamNameIs",
                () =>
```

File: src/libraries/System.Security.AccessControl/src/System/Security/AccessControl/ACL.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (MattKotsenas) [reply src/libraries/System.Security.AccessControl/src/System/Security/AccessControl/ACL.cs:590]: Sorry, I should have been clearer. Yes, I verified that both before and after my change I see the same `ArgumentOutOfRangeException`

File: src/libraries/System.Security.AccessControl/tests/RawAcl/RawAcl_RemoveAce.cs (hunk 2 above)
Comment (MattKotsenas) [reply src/libraries/System.Security.AccessControl/tests/RawAcl/RawAcl_RemoveAce.cs]: Good idea! Added!

**Review by bartonjs — APPROVED** [review]

**Comment by MattKotsenas** [comment]
Thanks @bartonjs! Unfortunately, I don't have merge access. Would you be able to complete this PR on my behalf? If you have any questions, please let me know. Thanks!

**Pull request merged**